### PR TITLE
Implement state dashboard refinements

### PR DESCRIPTION
**Pull request by mgwalker**
Addresses #722.

### This pull request changes...
- updates the dashboard
- hooks up the dashboard to the real list of APDs, at least nominally

### This pull request is ready to merge when...
- [ ] Tests have been updated (and all tests are passing)
- [ ] This code has been reviewed by someone other than the original author

### This feature is done when...
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: web/src/components/SidebarLink.js [hunk 1]
```diff
@@ -24,12 +24,24 @@ const getClasses = (depth, hash, anchor) => deline`
...
+const clickHandler = (depth, toggleExpand, propsOnClick) => e => {
+  if (depth === 0) {
+    toggleExpand();
+  }
+  if (propsOnClick) {
+    propsOnClick();
+    e.preventDefault();
+    e.stopPropagation();
+  }
+};
```
Comment (mgwalker) [inline web/src/components/SidebarLink.js]: This `depth` piece already existed, but the gist of it is that if this link is at the top of the tree, then a click should expand it to show the links underneath it.

The `propsOnClick` thing is new - the idea is that if the thing creating this link hooks up its own click handler, we'll use that and suppress the default click event.

File: web/src/components/StateDashboard.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+              events={[{ day: 7, title: 'Annual APD Due' }]}
+            />
+          </DashboardSection>
+        </div>
+      </div>
+    </div>
+  </div>
+);
+
+export default StateDashboard;
```
Comment (mgwalker) [inline web/src/components/StateDashboard.js]: Github says this is a new file, but this is just a renamed file with a bunch of changes, so...  I guess that's about the same?

Several of the components in here are essentially placeholders - they're not hooked up to any real data, but we want users to be able to see what the intent is, so we went ahead and built them out.  They're not necessarily super great (especially several components in one file 😬), but I expect they'll get split out over time as they get hooked up to real data.

File: web/src/containers/StateDashboardCurrentDocuments.js [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    case 'approved':
+      states.one.done = true;
+      states.two.done = true;
+      states.three.done = true;
+      actionButtonText = 'Approval letter';
+      break;
+
+    default:
+      break;
+  }
```
Comment (mgwalker) [inline web/src/containers/StateDashboardCurrentDocuments.js]: I'm really not wild about this `switch` statement to map a status into state info, but I wasn't sure offhand of a better way...

File: web/src/containers/StateDashboardCurrentDocuments.js [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+          <DocumentItem
+            key={apd.id}
+            name={`APD for ${apd.years.join(', ')}`}
+            status={apd.status}
+            buttonClick={this.open(apd.id)}
+          />
+        ))}
+        <DocumentItem name="Example one" status="reviewed" />
+        <DocumentItem name="Example two" status="in review" />
+        <DocumentItem name="Example three" status="approved" />
```
Comment (mgwalker) [inline web/src/containers/StateDashboardCurrentDocuments.js]: These are here to show off the various states that the progress bar could be in, for usability testing.

**Comment by mgwalker** [comment]
There's still work to do to get it up to Laura's design (e.g., the "approval letter" button needs to be a link; tables need to be sortable - or at least have a sorting dropdown that doesn't do anything; fonts need to be tweaked), but I thought this would be close enough we could move forward with it.  Let me know if it's not and I'll dive back in.  😄

**Comment by jeromeleecms** [comment]
@mgwalker If we merge this, do we need to spin up a separate issue for the content @lauraponce included in #722 this Friday? The link in #722 to the viewable version of the PR says it no longer exists so I can't check what it looks like...

**Comment by NAretakis** [comment]
Wait, is HiC-Bot talking to CMS's AWS?

**Comment by mgwalker** [comment]
Yep.

**Comment by mgwalker** [comment]
@jeromeleecms I think I can roll them into this.

File: web/src/containers/StateDashboardCurrentDocuments.js (hunk 3 above)
Comment (hbillings) [reply web/src/containers/StateDashboardCurrentDocuments.js]: The only thing that comes to mind is creating an object that maps these to the status data, like:
```
words = {
  'draft': {
    'actionButtonText': 'View',
    'actionButtonIcon': faEdit,
  }
}
```
and then you could just return something like `{ words[status].actionButtonText, words[status].actionButtonIcon }`. I'm not sure it's _better_, just different.

The other thing that might come in really handy is a `constants` file where you can define all your text. CALC does this [here](https://github.com/18F/calc/blob/develop/frontend/source/js/data-explorer/constants.js), and it's saved me a world of typo pain. (Also makes it _super_ easy when someone decides that text needs to be tweaked slightly.)

File: web/src/containers/StateDashboardSidebar.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+class Sidebar extends Component {
+  createApd = () => this.props.createApd();
+
+  pickApd = id => () => this.props.selectApd(id);
+
+  render() {
+    const { apds, place } = this.props;
+
+    return (
+      <div className="site-sidebar bg-teal relative">
```
Comment (hbillings) [inline web/src/containers/StateDashboardSidebar.js]: This should probably be an `aside` element.

File: web/src/containers/StateDashboardCurrentDocuments.js (hunk 3 above)
Comment (mgwalker) [reply web/src/containers/StateDashboardCurrentDocuments.js]: Oooh, I like it.  This button text will eventually come from a resource file, but the key can live in a constants file.  I'll make that change!

For the statuses, those are read-only from the API, so we could massage them into `Symbol`s also defined in a constants file, to make more guarantees about using them correctly elsewhere.  🤔  Probably shouldn't tack more onto this PR, but I love this as a future cleanup.  The statuses aren't used much right now, so good time to make that change.

File: web/src/containers/StateDashboardCurrentDocuments.js [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+  //     done: false,
+  //     started: false,
+  //     title: 'In Clearance'
+  //   },
+  //   five: {
+  //     done: false,
+  //     started: false,
+  //     title: 'Approved'
+  //   }
+  // };
```
[the remark this answers is not in the document]
Comment (mgwalker) [reply web/src/containers/StateDashboardCurrentDocuments.js]: Oopsie.  Yes!

File: web/src/components/StateDashboard.js [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+          (ex.) {e.title}
+        </div>
+      </div>
+    ))}
+  </div>
+);
+
+Events.propTypes = {
+  month: PropTypes.string.isRequired,
+  events: PropTypes.array.isRequired
```
[the remark this answers is not in the document]
Comment (mgwalker) [reply web/src/components/StateDashboard.js]: Interesting!  That makes total sense.  I wonder why we've never done it that way before.

[the remark this answers is not in the document]
Comment (mgwalker) [reply web/src/components/StateDashboard.js]: Oh, I see.  Our linter complains if we set `defaultProps` for required props.  Interesting...  https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/default-props-match-prop-types.md

**Pull request merged**